File: javascript/anagram/anagram.js
var Anagram = function (word) {
  this.target = this.toAnagramKey(word);
};

Anagram.prototype.match = function(words) { 
  return words.filter(this.isAnagram, this);
};

Anagram.prototype.toAnagramKey = function(word) { 
  return word.toLowerCase().split('').sort().join();
};

Anagram.prototype.isAnagram = function(word) { 
  return this.toAnagramKey(word) === this.target;
};

module.exports = Anagram;

File: clojure/anagram/anagram.clj
(ns anagram
  (:require [clojure.string :as string]))

(defn- to-anagram-key [word]
  (-> word string/lower-case sort ))

(defn- is-anagram [target word]
  (= (to-anagram-key target) (to-anagram-key word)))

(defn anagrams-for [target words] 
  (filter (fn [word]
              (is-anagram target word))
           words))
